pkg/minikube/bootstrapper/certs.go

```diff
@@ -38,10 +39,23 @@ var (
 		"ca.crt", "ca.key", "apiserver.crt", "apiserver.key", "proxy-client-ca.crt",
 		"proxy-client-ca.key", "proxy-client.crt", "proxy-client.key",
 	}
-	// This is the internalIP , the API server and other components communicate on.
-	internalIP = net.ParseIP(util.DefaultServiceClusterIP)
 )
 
+func internalIP(k8s KubernetesConfig) (net.IP, error) {
+	var svcNet net.IPNet
+	for _, extraOption := range k8s.ExtraOptions {
+		if extraOption.Component == "apiserver" && extraOption.Key == "ServiceClusterIPRange" {
+			ip, net, err := net.ParseCIDR(extraOption.Value)
+			if err != nil {
+				return ip, err
+			}
+			svcNet = *net
+		}
+	}
+	_, masterIP, err := master.DefaultServiceIPRange(svcNet)
```

Comment:
if its not obvious, this functions already returns `10.0.0.1` when given a nil argument:
https://github.com/kubernetes/minikube/blob/023aea13a24d1c8320d541974ed1d9d2fccb13ae/vendor/k8s.io/kubernetes/pkg/master/services.go#L33-L41

so it didn't seem worth adding an explicit else with `util.DefaultServiceClusterIP` here

but you might reasonable disagree